File: src/main/java/tech/bobliu/assignment08/model/Thread.java
package tech.bobliu.assignment08.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "threads")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Thread {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private Date datetime = new Date();

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User sender;

    @OneToMany(mappedBy = "thread", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Post> posts = new ArrayList<>();
}

File: src/main/kotlin/tech/bobliu/assignment08/model/Thread.kt
package tech.bobliu.assignment08.model

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "threads")
class Thread(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    var title: String = "",

    var datetime: Date = Date(),

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    var sender: User? = null,

    @OneToMany(mappedBy = "thread", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    var posts: MutableList<Post> = mutableListOf(),
)
